<?xml version="1.6" ?>

<launch>

    <!-- Node name -->
    <arg name="motor_node_name" default="default_wheels" />

    <!-- ROS topic name -->
    <arg name="sub_topic_name" default="default/wheel_command" />
    <arg name="emer_topic" default="none"/>

    <!-- USB ports names -->
    <arg name="back_port_name" default="/dev/ttyUSB0" />
    <arg name="front_port_name" default="/dev/ttyUSB1" />

    <!-- Robot side -->
    <arg name="robot_side" default="left" />

    <!-- Serial configs -->
    <arg name="serial_baud" default="57600" />
    <arg name="serial_parity" default="0" /> <!-- 0:None, 1:odd, 2:even -->
    <arg name="serial_stop_bit" default="1" /> <!-- 1:One, 2:Two, 3:One point five -->
    <arg name="serial_data_bits" default="8" /> <!-- 5, 6, 7 or 8 -->
    <arg name="serial_hdw_flow_ctrl" default="0" /> <!-- 0:None, 1:software, 2:hardware -->
    <arg name="serial_timeout" default="50" />
    
    
    <!-- The motor node -->
    <node name="$(arg motor_node_name)" type= "motor_node" pkg="engrais_motors" output="screen">
        <param name="sub_topic" type="string" value="$(arg sub_topic_name)" />

        <param name="back_port" type="string" value="$(arg back_port_name)" />
        <param name="front_port" type="string" value="$(arg front_port_name)" />
        <param name="emergency_topic" type="string" value="$(arg emer_topic)"/>

        <param name="side" type="string" value="$(arg robot_side)" />
        <param name="baud" type="int" value="$(arg serial_baud)" />
        <param name="parity" type="int" value="$(arg serial_parity)" /> <!-- 0:None, 1:odd, 2:even -->
        <param name="stop_bit" type="int" value="$(arg serial_stop_bit)" /> <!-- 1:One, 2:Two, 3:One point five -->
        <param name="data_bits" type="int" value="$(arg serial_data_bits)" /> <!-- 5, 6, 7 or 8 -->
        <param name="hdw_flow_ctrl" type="int" value="$(arg serial_hdw_flow_ctrl)" /> <!-- 0:None, 1:software, 2:hardware -->
        <param name="timeout" type="int" value="$(arg serial_timeout)" />
    </node>

</launch>
